app/scripts/metamask-controller.js

```diff
@@ -7474,4 +7517,82 @@ export default class MetamaskController extends EventEmitter {
   #getGlobalNetworkClientId() {
     return this.networkController.state.selectedNetworkClientId;
   }
+
+  #initControllers({ initObjects, initState }) {
```

Comment:
Both these functions could go in a separate `app/scripts/controller-init/utils.ts` for better encapsulation and testing, and TypeScript.